src/wallet/spend.cpp

```diff
@@ -832,11 +845,33 @@ static std::optional<CreatedTransactionResult> CreateTransactionInternal(
     // to avoid conflicting with other possible uses of nSequence,
     // and in the spirit of "smallest possible change from prior
     // behavior."
-    const uint32_t nSequence{coin_control.m_signal_bip125_rbf.value_or(wallet.m_signal_rbf) ? MAX_BIP125_RBF_SEQUENCE : CTxIn::MAX_SEQUENCE_NONFINAL};
+    bool use_anti_fee_sniping = true;
+    const uint32_t default_sequence{coin_control.m_signal_bip125_rbf.value_or(wallet.m_signal_rbf) ? MAX_BIP125_RBF_SEQUENCE : CTxIn::MAX_SEQUENCE_NONFINAL};
     for (const auto& coin : selected_coins) {
-        txNew.vin.push_back(CTxIn(coin.outpoint, CScript(), nSequence));
+        uint32_t sequence = default_sequence;
+        if (coin_control.HasSequence(coin.outpoint)) {
+            sequence = coin_control.GetSequence(coin.outpoint);
+            // If an input as a preset sequence, we can't do anti-fee-sniping
```

Comment:
Why not? As long as one input has a non-final sequence, it should still work.

(`DiscourageFeeSniping` would need its sanity checks adjusted to tolerate it, though)

Reply:
Where is this behaviour in the current logic? I thought presently it would do the anti-sniping, and then possibly (at worst) have that partially "broken" by subsequent changes?

Reply:
It's more of a side effect of `FundTransaction` rather than explicit behavior. Currently, having preset sequences inherently means a preset locktime which would disable anti-fee-sniping. This is just due to how `FundTransaction` works, and `FundTransaction` is the only way that preset sequences can even be provided.

Until we update `DiscourageFeeSniping` to handle preset sequences and locktime, I would prefer that we have this check explicitly just to avoid any problems in the future with how and when preset sequences and locktimes can be set.

Reply:
Making sure i got this one right. `use_anti_fee_sniping` may only be `false` if this is called from `FundTransaction`. And `FundTransaction` diregards the value of the `nLockTime` that may be set here anyways.

Reply:
Yes, `FundTransaction` currently throws away `nLockTime` thereby disabling anti-fee-sniping, and this preserves that behavior.